database/query/profile_status.sql

```diff
@@ -2,7 +2,7 @@
 -- name: UpsertRuleEvaluations :one
 INSERT INTO rule_evaluations (
     profile_id, repository_id, artifact_id, pull_request_id, rule_type_id, entity, rule_name
-) VALUES ($1, $2, $3, $4, $5, $6, sqlc.narg(rule_name))
+) VALUES ($1, $2, $3, $4, $5, $6, $7)
```

Comment:
It took me a bit to verify. this, so I'm going to record it.  I think this is safe because the migration script runs _before_ the server update.

Because migrations run before the server code update, we could have the server attempting to insert `rule_name == NULL` from https://github.com/stacklok/minder/blob/b7fb73a53159a74dd6307a4dd7de4c614ec8aef1/internal/engine/eval_status.go#L129 unless we've backfilled the table.  Once the table is backfilled, that shouldn't happen, and all the following rule evaluations should have name set:

Path:

* `EvalStatusParams` is set [here in `createEvalStatusParams`](https://github.com/stacklok/minder/blob/b7fb73a53159a74dd6307a4dd7de4c614ec8aef1/internal/engine/eval_status.go#L48) from the `Profile.Rule` and not otherwise.
* [`getEvaluator`](https://github.com/stacklok/minder/blob/main/internal/engine/executor.go#L272) is the only caller of `createEvalStatusParams`, which is [a pass-through from the `TraverseRules` in `evalEntityEvent`](https://github.com/stacklok/minder/blob/main/internal/engine/executor.go#L239).
* [`TraverseRules` runs the supplied function over each rule in the `relevant` set](https://github.com/stacklok/minder/blob/main/internal/engine/profile.go#L142) computed by [`GetRulesForEntity`](https://github.com/stacklok/minder/blob/main/internal/engine/profile.go#L101), which selects one of `repository`, `build_environment`, `pull_requests`, or `artifacts` from the lists of rules in the Profile and returns them without edit.
* We get the set of Profiles from [`ListProfileByProjectID`](https://github.com/stacklok/minder/blob/main/internal/engine/executor.go#L224) passed through [`MergeDatabaseListIntoProfiles`](https://github.com/stacklok/minder/blob/main/internal/engine/profile.go#L156)
* While `ListProfilesByProjectID` is auto-generated, `MergeDatabaseListIntoProfiles` [calls `rowInfoToProfileMap`](https://github.com/stacklok/minder/blob/main/internal/engine/profile.go#L187) to fill out the rules in the proto.
* `rowInfoToProfileMap` [unpacks the list of rules stored as JSON in the database](https://github.com/stacklok/minder/blob/main/internal/engine/profile.go#L250), and returns them, which doesn't set rule names if not already present.
* The migration is selecting the rows FOR UPDATE, which should hopefully block any attempts to evaluate rules while the migration is in progress.